Add floating chat launcher bubble to homepage (opens /chat as iframe overlay)

@claude

## Verified target
File: `src/worker.js` (main branch, 325035 bytes as of this issue).
- Homepage HTML is generated by `buildHomepageHtml` (confirmed present, byte offset ~135273 — grep for it, don't scan the whole file).
- `/chat` route confirmed present (grep `"/chat"`) — this is the full existing chat page (SSE text + Talk to Deed voice widget, already live, do not touch its internals).

## Problem
Right now the only way from the homepage to `/chat` is a plain text link in the footer nav. Real visitors don't find it — chat needs a floating launcher visible on the homepage.

## Task
1. In `buildHomepageHtml`, add a floating chat launcher bubble: fixed position, bottom-right, visible on scroll, brand navy+orange, small "Chat" label or icon. Match the site's existing visual language (see hero section styling in the same function for color vars / fonts).
2. On click: open a full-screen (mobile) / large centered panel (desktop) modal overlay containing an iframe pointing at /chat. Include a visible close button and close-on-backdrop-click.
3. Do NOT duplicate the /chat page's HTML/JS into the homepage — the iframe loads the real existing route as-is. This is intentional: single source of truth, zero drift risk between two chat implementations.
4. Iframe sizing: full viewport on mobile, roughly 90 percent width/height centered panel with rounded corners on desktop.
5. Bubble and overlay must not interfere with existing homepage CTAs (Get a Report, Start Investor, Start Pro) or the county-report email capture form already on the page.
6. English UI, matches existing site conventions (this repo has no build step — vanilla JS/CSS only, template literal HTML like the rest of the file).

## Do NOT
- Do not touch /chat route internals or the voice/SSE code already shipped there.
- Do not create a new deploy workflow — existing deploy-worker.yml deploys automatically on merge to main.
- Do not modify pricing, CTAs, or the county selector/email capture logic already on the homepage.

## Acceptance
PR with diff. Confirm in the PR description: bubble visible on page load, overlay opens on click, iframe loads /chat correctly, close button works, no layout shift/overlap with existing homepage elements. Note how you tested (no local dev server in this repo — say how you validated).

@claude Your PR #18339 broke the wrangler build. Reverted on main (commit 2b9f6d4, confirmed green again) — the live site was never actually affected since Cloudflare Workers only swaps in a NEW deploy after a successful build (old good version kept serving), but main needs a real fix.

## Exact error
```
✘ [ERROR] Expected ";" but found "html"

src/worker.js:4864:30:
  4864 │ const TERMS_HTML = `<!doctype html><html lang="en"><head>
       │                               ~~~~
       ╵                               ;
```

Line 4864 is `TERMS_HTML` — completely unrelated to your change. This is the classic signature of an **unbalanced backtick earlier in the file**: your new template literal (the bubble/overlay HTML) almost certainly has a stray, unescaped, or unclosed backtick, which merged everything after it into one giant string until it happened to re-sync (badly) at TERMS_HTML.

## Fix instructions
1. Re-pull latest main (already reverted, clean baseline — commit 2b9f6d4).
2. Re-apply your bubble/overlay change.
3. Before pushing, specifically check: any backtick ( ` ) character inside your new HTML template literal — including inside inline `<style>` or `<script>` blocks, or in any CSS content like `content: '...'` — must either not exist or be escaped as `\``.
4. Run a local syntax check before pushing if at all possible (e.g. `node --check src/worker.js` or `npx esbuild src/worker.js --bundle --outfile=/dev/null` to catch this class of error without needing a full wrangler deploy).
5. Confirm you ran that check and it passed, in your PR description this time — don't rely on CI to catch it first.

--- a/src/worker.js
+++ b/src/worker.js
@@ -4503,6 +4503,23 @@ footer{padding:2.5rem 2rem;background:var(--navy-band);border-top:1px solid var(
   .tile{padding:10px 12px}
   .feat-section{font-size:10px}
 }
+
+/* CHAT BUBBLE */
+#chat-bubble{position:fixed;bottom:24px;right:24px;z-index:9000;display:flex;align-items:center;gap:8px;background:var(--orange);color:var(--navy);border:none;border-radius:999px;padding:13px 20px;font-size:14px;font-weight:700;font-family:Inter,sans-serif;cursor:pointer;box-shadow:0 4px 24px rgba(249,115,22,.45);transition:background .15s,transform .15s;white-space:nowrap;letter-spacing:.02em}
+#chat-bubble:hover{background:var(--orange-hover);transform:translateY(-2px)}
+#chat-bubble svg{flex-shrink:0}
+
+/* CHAT OVERLAY */
+#chat-overlay{display:none;position:fixed;inset:0;z-index:9100;background:rgba(0,0,0,.65);backdrop-filter:blur(4px);-webkit-backdrop-filter:blur(4px);align-items:center;justify-content:center}
+#chat-overlay.open{display:flex}
+#chat-panel{position:relative;width:90vw;height:90vh;max-width:960px;background:var(--navy);border:1px solid var(--charcoal);border-radius:16px;overflow:hidden;display:flex;flex-direction:column;box-shadow:0 24px 80px rgba(0,0,0,.7)}
+#chat-panel iframe{flex:1;width:100%;border:none;background:var(--navy)}
+#chat-close{position:absolute;top:12px;right:14px;z-index:10;background:rgba(11,25,41,.85);border:1px solid var(--charcoal);color:var(--slate);border-radius:999px;width:34px;height:34px;font-size:18px;cursor:pointer;display:flex;align-items:center;justify-content:center;line-height:1;transition:background .15s,color .15s}
+#chat-close:hover{background:var(--charcoal);color:#fff}
+@media(max-width:767px){
+  #chat-panel{width:100vw;height:100dvh;max-width:none;border-radius:0;border:none}
+  #chat-bubble{bottom:16px;right:16px;padding:12px 16px;font-size:13px}
+}
 </style>
 </head>
 <body>
@@ -4805,6 +4822,40 @@ footer{padding:2.5rem 2rem;background:var(--navy-band);border-top:1px solid var(
     </div>
   </div>
 </footer>
+
+<!-- CHAT BUBBLE -->
+<button id="chat-bubble" onclick="openChat()" aria-label="Open chat">
+  <svg width="18" height="18" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2.5" stroke-linecap="round" stroke-linejoin="round" aria-hidden="true"><path d="M21 15a2 2 0 0 1-2 2H7l-4 4V5a2 2 0 0 1 2-2h14a2 2 0 0 1 2 2z"/></svg>
+  Chat
+</button>
+
+<!-- CHAT OVERLAY -->
+<div id="chat-overlay" role="dialog" aria-modal="true" aria-label="BidDeed chat" onclick="handleOverlayClick(event)">
+  <div id="chat-panel">
+    <button id="chat-close" onclick="closeChat()" aria-label="Close chat">&#x2715;</button>
+    <iframe id="chat-iframe" title="BidDeed Chat" allow="microphone"></iframe>
+  </div>
+</div>
+
+<script>
+function openChat(){
+  var overlay=document.getElementById('chat-overlay');
+  var iframe=document.getElementById('chat-iframe');
+  if(!iframe.src||iframe.src===window.location.href) iframe.src='/chat';
+  overlay.classList.add('open');
+  document.body.style.overflow='hidden';
+}
+function closeChat(){
+  document.getElementById('chat-overlay').classList.remove('open');
+  document.body.style.overflow='';
+}
+function handleOverlayClick(e){
+  if(e.target===document.getElementById('chat-overlay')) closeChat();
+}
+document.addEventListener('keydown',function(e){
+  if(e.key==='Escape') closeChat();
+});
+</script>
 
 ${HOMEPAGE_SCRIPT}
 </body>

